.github/workflows/cut-new-release.yml

```
@@ -0,0 +1,100 @@
+name: Cut a new release
+
+on:
+  workflow_dispatch:
+    inputs:
+      version:
+        type: string
+        description: Version of the next release (e.g. v0.15.0)
+        required: true
+
+env:
+  git-user: github-actions[bot]
+  git-email: 41898282+github-actions[bot]@users.noreply.github.com
+  main-branch: main
+
+jobs:
+  workflow-metadata:
+    name: Extract workflow metadata
+    runs-on: ubuntu-22.04
+    outputs:
+      ref: ${{ steps.extract.outputs.ref }}
+      base-version: ${{ steps.extract.outputs.base-version }}
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          token: ${{ secrets.GH_TOKEN }}
+          fetch-depth: 0
+
+      - name: Extract
+        id: extract
+        shell: bash
+        run: |
+          REF=$(echo "${{ env.main-branch }}")
+          # Valid version format
+          if [[ "${{ inputs.version }}" =~ ^v([0-9]+)\.([0-9]+)\.([0-9]+)$ ]]; then
+            # Find release branch from provided version. This resolves '0.14' from 'v0.14.0|1|2...'
+            BASE_VERSION=$(echo "${{ inputs.version }}" | cut -c2- |  awk 'BEGIN{FS=OFS="."}NF--')
+            echo "Finding branch: release-${BASE_VERSION}..." 
+            branch_exists=$(git ls-remote --heads origin release-${BASE_VERSION}) 
+            if [[ -z ${branch_exists} ]]; then
+              echo "release-${BASE_VERSION} is not found." 
+            else 
+              echo "release-${BASE_VERSION} is found, now REF=release-${BASE_VERSION}"
+              REF=$(echo "release-${BASE_VERSION}") 
+            fi 
+            echo ::set-output name=ref::$REF
+            echo ::set-output name=base-version::$BASE_VERSION
+          else
+            # Fail entire job if it is not a valid format
+            echo "${{ inputs.version }} is not a valid version format. Use something like 'v0.14.0'"
```

```suggestion
            echo "${{ inputs.version }} is not a valid version format. Expected pattern 'vMAJOR.MINOR.PATCH', e.g. `v0.14.0'"
```

BTW, should we expect here only `vMAJOR.MINOR.0` or just `vMAJOR.MINOR`?  As the patch doesn't make sens here. We will have patch releases only on "Finalize release", right?

---

We need full version, since I derive base versions from that. If you provide v0.14, it becomes tricky to understand if this is a patch release or not. However, if you provide v0.14.0 or v0.14.1, I easily understand what kind of release is that. `v0.14.0` this version is important for further calculations.

---

But we shouldn't use the `Cut new release` for doing the patch releases, right? so we always should use `v0.14.0`, `v0.15.0` etc, if we wont to do the patch, then we should run the `Finalize release` on a given branch e.g. `release-0.14` and there specify `v0.14.1`,  do I miss something?